backend/auth.ts

```diff
@@ -0,0 +1,72 @@
+import { PrismaAdapter } from '@lucia-auth/adapter-prisma';
+import type { IncomingMessage, ServerResponse } from 'http';
+import { Lucia, TimeSpan } from 'lucia';
+import prisma from '~/prisma';
+import type { CreateSession, User } from '~/shared/types';
+import type { Session } from 'lucia';
+
+export const adapter = new PrismaAdapter(prisma.session, prisma.user);
+
+export const lucia = new Lucia(adapter, {
+  sessionExpiresIn: new TimeSpan(14, 'd'),
+  sessionCookie: {
+    attributes: {
+      secure: process.env.NODE_ENV === 'production',
+    },
+  },
+  getUserAttributes({
+    createdAt,
+    email,
+    firstName,
+    lastName,
+    updatedAt,
+    uuid,
+  }: User): User {
+    return {
+      uuid: uuid,
+      firstName: firstName,
+      lastName: lastName,
+      email: email,
+      createdAt: createdAt,
+      updatedAt: updatedAt,
+    };
+  },
+});
+
+declare module 'lucia' {
+  interface Register {
+    Lucia: typeof lucia;
+    DatabaseUserAttributes: DatabaseUserAttributes;
+    DatabaseSessionAttributes: DatabaseSessionAttributes;
+  }
+}
+interface DatabaseUserAttributes extends User {}
+interface DatabaseSessionAttributes extends CreateSession {}
```

Comment:
Rivit **43** ja **44** näyttäisi olevan ihan turhat

```suggestion
declare module 'lucia' {
  interface Register {
    Lucia: typeof lucia;
    DatabaseUserAttributes: User;
    DatabaseSessionAttributes: CreateSession;
  }
}
```

Täytyy laittaa testiin ennenkuin tekee muutoksia, ettei mikään hajoa, mutta tämmöisen huomasin :smile: